### PR TITLE
Flutter SDK updates

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Automatic screen recording section added.
> Optimized the doc to remove the obviously unnecessary callouts.
> Fixed the language and improved clarity in some sections.

## Type of documentation

- [ ] New documentation
- [x] Documentation update
- [ ] Hotfix

### CODE REVIEW

**Review by desusai7 — APPROVED** [review]

File: docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx [hunk 1]
```diff
@@ -63,10 +58,11 @@ flutter pub get
 
 To install and use the Flutter SDK in your web app, follow the above steps to add the Flutter SDK. Additionally, you need to <Link to="/sources/event-streams/sdks/rudderstack-javascript-sdk/quick-start-guide/#step-1-install-javascript-sdk">install the RudderStack script</Link> in the `<head>` section of your web page.
 
-<div class="warningBlock">
+Not all the Flutter SDK API are applicable for the web. Calling the following API on your web app will have **no effect**:
```
Comment (ishachopra0401) [inline docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx:61]:
```suggestion
Not all the Flutter SDK APIs are applicable for the web. For example, the following APIs will have **no effect** on your web app :
```

File: docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx [hunk 2]
```diff
@@ -12,39 +13,33 @@ The **RudderStack Flutter SDK** lets you track event data from your Flutter appl
...
-<div class="successBlock">
-Starting from v1.0.2, the RudderStack Flutter SDK is migrated to Null Safety.
-</div>
+You will also need the data plane URL. Refer to the <Link to="/dashboard-guides/overview/#data-plane-url">Dashboard Overview</Link> guide for more information on the data plane URL and where to find it.
 
 ## Installing the RudderStack Flutter SDK
 
-The recommended way to install the Flutter SDK is through [`pub`](https://pub.dev/packages/rudder_sdk_flutter).
+<div class="infoBlock">
+Starting from v1.0.2, the RudderStack Flutter SDK is migrated to Null Safety.
```
Comment (ishachopra0401) [inline docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx:39]: Lil confused.. Google says Null Safety is a feature but here it sounds like a package manager like maven central, nuget or pub.

File: docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx [hunk 3]
```diff
@@ -97,12 +93,17 @@ The `initialize` method has the following signature:
 | `writeKey` | String       | Required      | Your Flutter source write key.                       |
 | `config`   | `RudderConfig` | Optional       | Contains the RudderStack client configuration. |
 
-<div class="infoBlock">
+## Self-hosted control plane
+
+For the <Link to="/destinations/rudderstack-connection-modes/#device-mode">device mode</Link>-supported destinations, the Flutter SDK needs to fetch the required configuration from the control plane.
```
Comment (ishachopra0401) [inline docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx:98]: Does this mean that this section is only relevant for device mode-supported destinations?

File: docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx [hunk 4]
```diff
@@ -97,12 +93,17 @@ The `initialize` method has the following signature:
 | `writeKey` | String       | Required      | Your Flutter source write key.                       |
 | `config`   | `RudderConfig` | Optional       | Contains the RudderStack client configuration. |
 
-<div class="infoBlock">
+## Self-hosted control plane
+
+For the <Link to="/destinations/rudderstack-connection-modes/#device-mode">device mode</Link>-supported destinations, the Flutter SDK needs to fetch the required configuration from the control plane. 
 
-Refer to the <a href="#configuring-the-rudderstack-client">Configuring your RudderStack client</a>  section below for a complete list of the configurable parameters.
+If you are self-hosting RudderStack and using the <a href="https://www.rudderstack.com/docs/rudderstack-open-source/control-plane-lite/">Control Plane Lite</a> utility to host your own control plane, then follow the steps in [this section](https://www.rudderstack.com/docs/rudderstack-open-source/control-plane-lite/#using-sdk-sources-set-up-in-self-hosted-control-plane) and specify `controlPlaneUrl` in your [`RudderConfig.Builder`](#sdk-initialization-options) that points to the hosted source configuration file.
```
Comment (ishachopra0401) [inline docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx:100]:
```suggestion
If you are self-hosting RudderStack using the <a href="https://www.rudderstack.com/docs/rudderstack-open-source/control-plane-lite/">Control Plane Lite</a> utility, then follow the steps in [this section](https://www.rudderstack.com/docs/rudderstack-open-source/control-plane-lite/#using-sdk-sources-set-up-in-self-hosted-control-plane) and specify the `controlPlaneUrl` in your [`RudderConfigBuilder`](#sdk-initialization-options) that points to the hosted source configuration file.
```

File: docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx [hunk 5]
```diff
@@ -465,17 +489,16 @@ rudderClient.identify("testUserId", options: option);
 
 ## Setting your own `anonymousId` using `putAnonymousId`
 
-By default, RudderStack uses `deviceId` as `anonymousId`. You can use the following method to override and set your own `anonymousId` with the SDK, as shown:
+By default, RudderStack uses the `deviceId` as the user's `anonymousId`. Use the following method to override and set your own `anonymousId` with the SDK:
```
Comment (ishachopra0401) [inline docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx:492]:
```suggestion
By default, RudderStack uses the `deviceId` as the user's `anonymousId`. Use the following method to override and set your own `anonymousId`:
```

File: docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx [hunk 6]
```diff
@@ -501,47 +524,23 @@ rudderClient.putAdvertisingId(<ADVERTISING_ID>);
 ```
 
 <div class="infoBlock">
-
-  The <code class="inline-code">id</code> parameter that you pass in the <code class="inline-code">putAdvertisingId</code> method is assigned as the AAID if you are on an Android device, and as the IDFA if you are on an iOS device.
+The <code class="inline-code">id</code> parameter that you pass in the <code class="inline-code">putAdvertisingId</code> method is assigned as the AAID if you are on an Android device, and as the IDFA if you are on an iOS device.
```
Comment (ishachopra0401) [inline docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx:527]: IMO, we can skip this infoblock as its already clear from the above content that for Android its AAID and for iOS its IDFA.
WDYT?

File: docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx [hunk 7]
```diff
@@ -582,9 +581,6 @@ Map context = await rudderClient.getRudderContext();
 print(context["traits"]);
 ```
 
-<div class="infoBlock">
-
-For web apps, the <code class="inline-code">getRudderContext</code> API returns the user traits and anonymous ID only.
-</div>
-
+For web apps, the `getRudderContext`API returns the user traits and anonymous ID only.
```
Comment (ishachopra0401) [inline docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx:584]:
```suggestion
For web apps, the `getRudderContext` API returns the user traits and anonymous ID only.
```
Also, acc to my understanding, the `getRudderContext`API:
* Returns **user traits** after making an identify call
* Returns **user traits** and **anonymous ID** for web apps

then why do we say:
>For web apps, the getRudderContextAPI returns the user traits and anonymous ID **only**.

Rather, it can be 
>For web apps, the getRudderContextAPI returns the user traits and anonymous ID **as well.**

LMK if I am missing anything

File: docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx (hunk 2 above)
Comment (ameypv-rudder) [reply docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx:39]: It's this feature: https://www.section.io/engineering-education/null-safety-support-for-flutter-and-dart/#:~:text=Null%20safety%20means%20that%20a,are%20shown%20at%20compile%2Dtime. cc @desusai7 for confirmation.

Comment (ameypv-rudder) [reply docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx:39]: maybe we can reword it to something like:

```
Starting from v1.0.2, the RudderStack Flutter SDK supports the <a href="https://dart.dev/null-safety">Null Safety</a> feature.
```

File: docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx (hunk 3 above)
Comment (ameypv-rudder) [reply docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx:98]: Nope, it's for the open source deployments where even the control plane is hosted. In that case, the SDK needs to know where the source-destination config file is hosted.

File: docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx (hunk 7 above)
Comment (ameypv-rudder) [reply docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx:584]: You're right - made the change accordingly.

File: docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx (hunk 6 above)
Comment (ameypv-rudder) [reply docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx:527]: Makes sense!

File: docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx [hunk 8]
```diff
@@ -212,19 +205,11 @@ Map context = await rudderClient.getRudderContext();
 print(context["traits"]);
 ```
 
-<div class="infoBlock">
-
-For web apps, the <code class="inline-code">getRudderContext</code> API returns the user traits and anonymous ID only.
-</div>
+For the web apps, the `getRudderContext` API returns the user's `traits` and the `anonymousId` only.
```
Comment (ishachopra0401) [inline docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx:208]:
```suggestion
For the web apps, the `getRudderContext` API returns the user's `traits` and the `anonymousId`.
```

File: docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx (hunk 2 above)
Comment (desusai7) [reply docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx]: yes, it's like a standard or a feature, which we were not adhering to till version 1.0.2.

And also the term migrating to null safety makes sense because we need to make many changes in the code base to be null safety compatible. 

Refer to the Flutter official doc for Migrating to Null Safety here https://dart.dev/null-safety/migration-guide

File: docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx (hunk 3 above)
Comment (ishachopra0401) [reply docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx]: understood.. then why do we say "for device mode-supported destinations"?

**Review by ishachopra0401 — APPROVED** [review]

File: docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx (hunk 2 above)
Comment (ameypv-rudder) [reply docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx]: Let's use the original "...migrated to Null Safety." only, then

File: docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx (hunk 3 above)
Comment (ameypv-rudder) [reply docs/sources/event-streams/sdks/rudderstack-flutter-sdk/flutter-v2.mdx]: Makes sense - removed the sentence entirely to avoid any confusion.
